libs/dissect/src/test/java/org/elasticsearch/dissect/DissectParserTests.java

```diff
@@ -375,13 +377,11 @@ private void assertMatch(String pattern, String input, List<String> expectedKeys
 
     private void assertMatch(String pattern, String input, List<String> expectedKeys, List<String> expectedValues, String appendSeperator) {
         Map<String, String> results = new DissectParser(pattern, appendSeperator).parse(input);
-        List<String> foundKeys = new ArrayList<>(results.keySet());
-        List<String> foundValues = new ArrayList<>(results.values());
-        Collections.sort(foundKeys);
-        Collections.sort(foundValues);
-        Collections.sort(expectedKeys);
-        Collections.sort(expectedValues);
-        assertThat(foundKeys, Matchers.equalTo(expectedKeys));
-        assertThat(foundValues, Matchers.equalTo(expectedValues));
+        assertThat(results.size(), Matchers.equalTo(expectedKeys.size()));
+        assertThat(results.size(), Matchers.equalTo(expectedValues.size()));
+        for (int i = 0; i < results.size(); i++) {
+            final String key = expectedKeys.get(i);
+            assertThat(results.get(key), Matchers.equalTo(expectedValues.get(i)));
+        }
```

Comment:
This slightly changes the assertion by no longer explicitly asserting that that the keys match. Is this change needed to fix the test(s) ?

Reply:
It does check that the keys match, given that the keys are unique in the **expectedKeys** array. The original check was flawed because it was sorting parallel arrays independently.